lib/packages/docutils/rst.nim

```diff
@@ -514,10 +525,17 @@ proc defaultFindFile*(filename: string): string =
   if fileExists(filename): result = filename
   else: result = ""
 
+proc defaultRoleKind(options: RstParseOptions): RstNodeKind =
```

Comment:
instead of adding `defaultRoleKind`, you could infer it from currRole, eg:
```
p.s.currRoleKind = defaultRoleKind(p.s.options)
p.s.currRole = defaultRole(p.s.options)

=>

p.s.currRole = defaultRole(p.s.options)
p.s.currRoleKind = whichRole(p, p.s.currRole)
```

(a bit more DRY)

Reply:
done.